src/main/scala/scala/async/internal/AnfTransform.scala

```diff
@@ -16,16 +16,18 @@ private[async] trait AnfTransform {
   import c.internal._
   import decorators._
 
-  def anfTransform(tree: Tree): Block = {
+  def anfTransform(tree: Tree, owner: Symbol): Block = {
     // Must prepend the () for issue #31.
-    val block = c.typecheck(atPos(tree.pos)(Block(List(Literal(Constant(()))), tree))).setType(tree.tpe)
+    val block = c.typecheck(atPos(tree.pos)(newBlock(List(Literal(Constant(()))), tree))).setType(tree.tpe)
 
     sealed abstract class AnfMode
     case object Anf extends AnfMode
     case object Linearizing extends AnfMode
 
+    val tree1 = adjustTypeOfTranslatedPatternMatches(block, owner)
```

Comment:
iuc, this changes the types before the trees are adjusted by eliminateLabelParameters - maybe explain that in a comment